src/core/net/ip6.cpp

```diff
@@ -1458,7 +1459,10 @@ void Ip6::UpdateBorderRoutingCounters(const Header &aHeader, uint16_t aMessageLe
     if (aIsInbound)
     {
         VerifyOrExit(!Get<Netif>().HasUnicastAddress(aHeader.GetSource()));
-
+        if (!aHeader.GetSource().MatchesPrefix(aHeader.GetDestination().GetPrefix().m8, kPrefixLength))
+        {
+            counter = &mBorderRoutingCounters.mInboundInternet;
+        }
```

Comment:
Not sure if checking first 48 bit of src/dest prefix match  is a good way?

Reply:
I'm refering to [this chapter](https://www.ripe.net/publications/docs/ripe-690#:~:text=Each%20hexadecimal%20character%20in%20an,site%20will%20be%20a%20%2F64)  to assert the /48 Internet address